### PR TITLE
Use Cell ID returned by the server

### PR DESCRIPTION
**Pull request by lukevmorris**
- [x] CHANGELOG.md updated
  - [x] Rebased/mergable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### The problem
Seen below. A cell element's `key` was calculated anew per `LayoutRenderer.render()` execution as we mapped over the dashboard's cells. This means the `key` was not guaranteed to be an invariant. And so if the key value changes, then the `GridLayout` component may decide to reposition more than it has to.
before

### The Solution
Each cell has an ID on the server -- use this value instead of the cell's index in the array. This should be relatively more invariant.
after

### CODE REVIEW

**Comment by cryptoquick** [comment]
Cool, is this related to the discussion we had earlier, last month? Do you think ID is used elsewhere, or was the code written in such a way that it's flexible enough to use field i, once as the index, now is the id? If so, that's very clever.

File: chronograf.go [hunk 1]
```diff
@@ -380,7 +380,7 @@ type Dashboard struct {
 
 // DashboardCell holds visual and query information for a cell
 type DashboardCell struct {
-	ID      string           `json:"-"`
+	ID      string           `json:"i"`
```
Comment (goller) [inline chronograf.go]: @lukevmorris  is the cell id no longer used?

Comment (lukevmorris) [inline chronograf.go]: The cell id is _now being_ used! I wanted to expose it as `i` in the cell json response. Is this the right thing to do? I just tried it out and it worked.

**Comment by lukevmorris** [comment]
But back to our discussion earlier this month. Now that we're passing the actual cell ID to/from the server, we can start to rephrase our reducers

File: chronograf.go (hunk 1 above)
Comment (goller) [inline chronograf.go]: You could do it that way, but, I'd rather see links.self used instead.

Comment (lukevmorris) [inline chronograf.go]: I tried to use `links.self` but it made the `GridLayout` all glitchy. Maybe I had something not quite right -- i'll try again.

**Comment by cryptoquick** [comment]
@jaredscheib encapsulated our thoughts in #1028 
We'll probably have to refer back to that

File: chronograf.go (hunk 1 above)
Comment (lukevmorris) [inline chronograf.go]: ok got it -- reverting

Comment (timraymond) [inline chronograf.go]: Agree w/ @goller here, we should try to use `links.self`. `react-grid-layout` is _very_ finicky about that `i` param though... you might need to go out of your way to coerce that into something it likes.

Comment (lukevmorris) [inline chronograf.go]: Yeah actually -- I noticed that it works with strings like `"abcdef-123456-abcdef-123456"` but not with strings like `dashboards/1/cells/abcdef ...`. I'm sure it's not the string that's the problem, but some sort of comparison that's happening in some component somewhere. So turns out my reversion doesn't actually work (which I never pushed).

**Comment by lukevmorris** [comment]
@nhaugo @timraymond @goller See: https://github.com/influxdata/chronograf/pull/1222#discussion_r110464144

`cells.links.self` doesn't work out of the box. Things get all glitchy, and we send an infinite stream of update requests to the server for that dashboard. So I didn't push it.

This PR works in its current state. Do we want to merge this and migrate toward `links.self` in beta-9, or hold off on merging this until we can get `links.self` working? Pro for merging it now is that the experience in beta-8 will be smooth. Pro for merging it later is that we won't have `id` code sitting around in a transitional state.

**Comment by timraymond** [comment]
I know that messing around with the ID can get hairy, so we should probably merge and add an issue for cleaning it up later. I'd say it's good tech debt.

**Pull request merged**